### PR TITLE
Update k8s and dependencies to v1.30.3

### PR DESCRIPTION
**Pull request by kishen-v**
**What type of PR is this?**

**What this PR does / why we need it**:
This PR contains
1. changes to update k8s and the associated dependencies to 1.30.3.
2. Removes `cmd` binary.
3. Mention go patch version in go.mod, to address the dependency of toolchain version requirement. - https://github.com/golang/go/issues/62278#issuecomment-1693538776

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:
The k8s-related modules and the test modules were individually updated using `go get -u`

**Release note**:
```
none
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module sigs.k8s.io/ibm-powervs-block-csi-driver
 
-go 1.22
+go 1.22.0
+
+toolchain go1.22.3
```
Comment (yussufsh) [inline go.mod:5]: Can you please delete this line? This creates an issue for sec scans in one of the downstream projects.

**Review by yussufsh — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (kishen-v) [reply go.mod]: Done, thanks!

**Review by yussufsh — APPROVED** [review]
/lgtm

**Review by yussufsh — APPROVED** [review]
/lgtm

File: tests/e2e/testsuites/testsuites.go [hunk 2]
```diff
@@ -474,7 +474,7 @@ func (t *TestPod) Cleanup() {
 }
 
 func (t *TestPod) WaitForSuccess() {
-	err := e2epod.WaitForPodSuccessInNamespaceSlow(context.TODO(), t.client, t.pod.Name, t.namespace.Name)
+	err := e2epod.WaitForPodSuccessInNamespaceTimeout(context.TODO(), t.client, t.pod.Name, t.namespace.Name, 15*time.Minute)
```
Comment (kishen-v) [inline tests/e2e/testsuites/testsuites.go:477]: `WaitForPodSuccessInNamespaceSlow` is deprecated from 1.30 - https://github.com/kubernetes/kubernetes/pull/122823.

**Pull request merged**